internal/config/config.go

```diff
@@ -360,11 +363,16 @@ func New(fileHandler file.Handler, name string, force bool) (*Config, error) {
 	}
 
 	// Read secrets from env-vars.
-	clientSecretValue := os.Getenv(constants.EnvVarAzureClientSecretValue)
-	if clientSecretValue != "" && c.Provider.Azure != nil {
+	clientSecretValue, ok := os.LookupEnv(constants.EnvVarAzureClientSecretValue)
+	if ok && c.Provider.Azure != nil {
```

Comment:
What's the reason for this change? If I understand it right, you are now allowing to explicitly set the clientSecretValue to the empty string, right?

Reply:
I was looking for a more general pattern in case we want to use this for other config options (where we may want an empty string). For the environment variables we currently use for overriding we could disallow the empty string.